src/java/voldemort/server/niosocket/AsyncRequestHandler.java

```diff
@@ -187,10 +247,7 @@ protected void write(SelectionKey selectionKey) throws IOException {
 
         // If we don't have anything else to write, that means we're done with
         // the request! So clear the buffers (resizing if necessary).
-        if(outputStream.getBuffer().capacity() >= resizeThreshold)
-            outputStream.setBuffer(ByteBuffer.allocate(socketBufferSize));
-        else
-            outputStream.getBuffer().clear();
+        outputStream.clear();
```

Comment:
So, this path will shrink the buffer in case there was a request greater than the upperBound, correct?  So, while getting rid of the second buffer is a win in terms of memory footprint (still BDB is going dominate the heap nonetheless anyways).. This growing and shrinking is the root cause of the mythical comm buffer problem. It boils down to the upperSizeBound is incorrect and we need to do some heap size checking to see if we can tolerate this value of not.. (eg:  say lowerbound is 64kb and higher is 128kb and request is 129kb, when you clear(), the bufferContainer(),reset() will reset this back to 64kb right?  (Am I getting this correct?).. Imagine this happening in a loop with lots of connections.. that problem still remains I think..